### PR TITLE
Fix/invalid config address with hornet v1.0.4

### PR DESCRIPTION
**Pull request by peterokwara**
# Description of change

When trying to run a private tangle with hornet v1.0.4, an invalid peer address error shows up
```
he following plugins are enabled: Coordinator RestAPI URTS
panic: invalid config peer address at pos 0: failed to parse multiaddr "/dns/node1/tcp/15600/p2p/G9yo8aUBbZFEgRWgkg8PZe2wqRi2nTm6KvUyQpt4KXFZ": invalid value "G9yo8aUBbZFEgRWgkg8PZe2wqRi2nTm6KvUyQpt4KXFZ" for protocol p2p: failed to parse p2p addr: G9yo8aUBbZFEgRWgkg8PZe2wqRi2nTm6KvUyQpt4KXFZ selected encoding not supported

goroutine 1 [running]:
github.com/gohornet/hornet/core/p2p.provide.func3(0x0, 0x0, 0xc000184760, 0x1e7846d, 0xc, 0x0)
        /app/core/p2p/core.go:173 +0x6a5
reflect.Value.call(0x1bb5b60, 0x238fba8, 0x13, 0x1e6d176, 0x4, 0xc0005181c8, 0x1, 0x1, 0x30, 0x7fb52c2d6108, ...)
        /usr/local/go/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0x1bb5b60, 0x238fba8, 0x13, 0xc0005181c8, 0x1, 0x1, 0x1cb7b20, 0x1dd2901, 0xc000508450)
        /usr/local/go/src/reflect/value.go:337 +0xb9
go.uber.org/dig.defaultInvoker(0x1bb5b60, 0x238fba8, 0x13, 0xc0005181c8, 0x1, 0x1, 0xc000652aa0, 0xc0005181c8, 0x1)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/dig.go:284 +0x65
go.uber.org/dig.(*node).Call(0xc0000b9b80, 0x2562058, 0xc000652aa0, 0x256f448, 0x1cc4500)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/dig.go:710 +0x27b
go.uber.org/dig.paramSingle.Build(0x0, 0x0, 0x0, 0x256f448, 0x1cc4500, 0x2562058, 0xc000652aa0, 0x1e430c0, 0xc000049e70, 0xc00052a050, ...)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/param.go:245 +0x2df
go.uber.org/dig.paramObjectField.Build(...)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/param.go:384
go.uber.org/dig.paramObject.Build(0x256f448, 0x1d7d8a0, 0xc0002f0c80, 0x5, 0x8, 0x2562058, 0xc000652aa0, 0xc000306b28, 0x5724bb, 0xc000101000, ...)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/param.go:311 +0x152
go.uber.org/dig.paramList.BuildList(0x256f448, 0x1b94cc0, 0xc0004f1840, 0x1, 0x1, 0x2562058, 0xc000652aa0, 0x0, 0x0, 0x1, ...)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/param.go:201 +0xb8
go.uber.org/dig.(*Container).Invoke(0xc000652aa0, 0x1b94cc0, 0x238fb90, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/dig.go:432 +0x2cb
github.com/gohornet/hornet/pkg/node.(*Node).init.func7(0xc00047e820, 0x1)
        /app/pkg/node/node.go:187 +0x6e
github.com/gohornet/hornet/pkg/node.forEachCorePlugin(0xc0000c9880, 0x9, 0x10, 0xc000221d78)
        /app/pkg/node/node.go:320 +0x51
github.com/gohornet/hornet/pkg/node.(*Node).ForEachCorePlugin(...)
        /app/pkg/node/node.go:328
github.com/gohornet/hornet/pkg/node.(*Node).init(0xc0003a39d0)
        /app/pkg/node/node.go:185 +0x77e
github.com/gohornet/hornet/pkg/node.New(0xc00069ff60, 0x3, 0x3, 0x1c7fca0)
        /app/pkg/node/node.go:43 +0x20f
github.com/gohornet/hornet/pkg/node.Run(0xc00069ff60, 0x3, 0x3, 0xc0004facc0)
        /app/pkg/node/node.go:62 +0x3f
main.main()
        /app/main.go:33 +0x334
The following plugins are disabled: Autopeering MQTT Profiling Prometheus Spammer WarpSync ZMQ
```
From https://github.com/iotaledger/one-click-tangle/blob/a8ff9269b76fd7f3eb1e4ef95426ca8fc263e52b/hornet-private-net/utils.sh#L21

Trying to generate a p2pIdentity with hornet v1.0.3, with the docker compose file having a hornet image with v1.0.3 `docker-compose run --rm node hornet tool p2pidentity > ~/test1.txt` we get the following result, which is 3 lines. The p2pPeerId is located at line 3.

```
Your p2p private key:  7895448bfb445af4726ec866d71ca2bc20a4d00db19b6f4ba4984f37c6bb4e62da7028a1f203c2e12738fb212837da502d319632161818df8cb025bb2d9f669b
Your p2p public key:  da7028a1f203c2e12738fb212837da502d319632161818df8cb025bb2d9f669b
Your p2p PeerID:  12D3KooWQX4Eur46dHdomZ5serSTQCxXB5t52NtpXxsA9Sjo2qZL
```

Trying to generate a p2pIdentity with hornet v1.0.4, with the docker compose file having a hornet image with v1.0.4 `docker-compose run --rm node1 hornet tool p2pidentity > ~/test2.txt` we get the following result, which is 4 lines. The p2pPeerId is now located at line 4. There is extra information as the p2p public key is provided both in hex and base58.

```
Your p2p private key (hex):  bf0cb199ccffc6ea99717608b1d096b3ca85dfbba103bc73a95163aa25e9861766c01b8a94b0d4906311b4fb14998186feb6bbf51d57577b672cc6a3d26488e1
Your p2p public key (hex):  66c01b8a94b0d4906311b4fb14998186feb6bbf51d57577b672cc6a3d26488e1
Your p2p public key (base58):  7v6VCDRWmuZphYdz241vvNDe2GzBQXxF8KEJqbiZ3Bgt
Your p2p PeerID:  12D3KooWGjTewM1DxoqK7S9QeemGhAJUWc2pSE2j5ubLDasSw8Qp
```

This brings problems when we try to extract the p2pPeerId here https://github.com/iotaledger/one-click-tangle/blob/a8ff9269b76fd7f3eb1e4ef95426ca8fc263e52b/hornet-private-net/utils.sh#L33 since  it will grab the p2p public key (base58).

This PR attempts to fix this problem, first by bumping the hornet version to v1.0.4. And then instructing the **getPeerID** function to fetch it from the 4th line.

This fix only works for hornet v1.0.4 Attempting to roll back to older v1.0.3 with this changes will result in things not working, as it will fetch the peerId from line 4 which will be inexistent/blank line. 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

1. Clone the project https://github.com/iotaledger/one-click-tangle
2. Modify the docker-compose file here https://github.com/iotaledger/one-click-tangle/blob/chrysalis/hornet-private-net/docker-compose.yml https://github.com/iotaledger/one-click-tangle/blob/chrysalis/hornet-private-net/extra-nodes/docker-compose.yml to fetch hornet v1.0.4
3. Modify https://github.com/iotaledger/one-click-tangle/blob/a8ff9269b76fd7f3eb1e4ef95426ca8fc263e52b/hornet-private-net/utils.sh#L21 number 3 to 4 so as to fetch peer id from the 4th line.
4. Run the private tangle by running `./private-tangle install 30` in the one-click-tangle/hornet-private-net folder. The node should be up and running without any problems whatsoever.
5. Upload the `full_snapshot.bin` temporarily to cloud (located in the one-click-tangle/hornet-private-net/snapshots/private-tangle folder) using the command `curl --upload-file ./full_snapshot.bin https://transfer.sh/full_snapshot.bin`. The command  provides a download url from which one can download the snapshot.
6. Move to the one-click-tangle/hornet-private-tangle/extra-nodes and download the snapshot file by running `curl  http://url_from_the_5th_step -o full_snapshot.bin`
7. Set up an extra node by running `private-hornet.sh [install|update|start|stop] <node_connection_str> <coo_public_key>? <peer_multiAdress>? <snapshot_file>?` Once it's up check if the extra node connects with the main node.

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by jmcanterafonseca-iota** [comment]
instead of breaking previous Hornet versions we can count the number of lines in the file with

```
cat identity.txt | wc -l | tr -d ' ' | tr -d '\n' | tr -d '\r'
```

and do the peerID extraction accordingly ...

**Review by jmcanterafonseca-iota — CHANGES_REQUESTED** [review]
let's keep backwards compatibility

**Comment by peterokwara** [comment]
Lemme try this out

**Comment by peterokwara** [comment]
Should the hornet version in the docker-compose file still be bumped up from hornet v1.0.3 to hornet v1.0.4?

**Comment by peterokwara** [comment]
Tested the script (manually via editing the docker-compose file and using the update script), it seems to be working fine with both hornet v1.0.3 and hornet v1.0.4

**Comment by peterokwara** [comment]
Reverted back to hornet v1.0.3

File: hornet-private-net/utils.sh [hunk 1]
```diff
@@ -30,5 +30,10 @@ setupIdentityPrivateKey () {
 # Extracts the peerID from the identity file
 getPeerID () {
   local identity_file="$1"
-  echo $(cat $identity_file | sed '3q;d' | cut -d ":" -f 2 | sed "s/ \+//g" | tr -d "\n" | tr -d "\r")
+  local line_count=$(cat $identity_file | wc -l | tr -d ' ' | tr -d '\n' | tr -d '\r')
+  if [ $line_count == "6" ]; then
```
Comment (jmcanterafonseca-iota) [inline hornet-private-net/utils.sh:36]:
```suggestion
local line_number=4
  if [ $line_count == "6" ]; then
    line_number=3
  fi
  echo $(cat $identity_file | sed '$line_numberq;d' | cut -d ":" -f 2 | sed "s/ \+//g" | tr -d "\n" | tr -d "\r")
```

**Review by jmcanterafonseca-iota — CHANGES_REQUESTED** [review]
there is a more elegant / concise way of doing it, please check.

**Review by jmcanterafonseca-iota — APPROVED** [review]
LGTM, thanks

**Pull request merged**